### PR TITLE
Simplify logic in TaskSnapshot component #8467 (PR)

### LINKED ISSUE #8467: Simplify logic in TaskSnapshot component
image

There are three core pieces of a single task in the `TaskSnapshot` component: chronological information, narrative information, and the actions menu. The first two pieces that hold information have grown a bit unwieldy over the past few months and are due for a refactor. The information that those two sections should hold are as follows.

#### Chronological information
* Assigned on date
* Days waiting (or days on hold if the task is on hold)
* Due date

#### Narrative information
* Assigned to
* Assigned by
* Prepared by (this information may already be shown as the attorney field in the case title, if it will always be the same let's not show it down here)
* Type label
* Instructions

## Acceptance criteria
* [ ] Display information about task in appropriate sections of the task snapshot.
* [ ] Use the `isLegacy` attribute of the task instead of the `isLegacyAppeal` attribute of the appeal to determine behaviour where possible.

### PR DESCRIPTION
**Pull request by rkreyhsig**
Resolves #8467 - Simplify logic in TaskSnapshot component

### CODE REVIEW

File: client/app/queue/TaskSnapshot.jsx [hunk 1]
```diff
@@ -207,19 +204,23 @@ export class TaskSnapshot extends React.PureComponent<Props> {
...
-              <dd><OnHoldLabel task={task} /></dd>
-            </React.Fragment>
-          }
+          <dt>{COPY.TASK_SNAPSHOT_TASK_TYPE_LABEL}</dt><dd>{CO_LOCATED_ADMIN_ACTIONS[task.label]}</dd>
           { task.instructions && task.instructions.length > 0 &&
-            <React.Fragment>
-              <dt>{COPY.TASK_SNAPSHOT_TASK_INSTRUCTIONS_LABEL}</dt>
-              <dd>{this.taskInstructionsWithLineBreaks(task.instructions)}</dd>
-            </React.Fragment> }
+            <div>
```
Comment (rkreyhsig) [inline client/app/queue/TaskSnapshot.jsx]: Task instructions link had not been added for legacy cases

File: client/app/queue/TaskSnapshot.jsx [hunk 2]
```diff
@@ -207,19 +204,23 @@ export class TaskSnapshot extends React.PureComponent<Props> {
...
+                <dd>{this.taskInstructionsWithLineBreaks(task.instructions)}</dd>
+              </React.Fragment> }
+              <Button
+                linkStyling
+                styling={css({ padding: '0' })}
+                name={this.state.taskInstructionsIsVisible ? COPY.TASK_SNAPSHOT_HIDE_TASK_INSTRUCTIONS_LABEL :
+                  COPY.TASK_SNAPSHOT_VIEW_TASK_INSTRUCTIONS_LABEL}
+                onClick={this.toggleTaskInstructionsVisibility} />
+            </div>
+          }
```
Comment (rkreyhsig) [inline client/app/queue/TaskSnapshot.jsx]: Task instructions link had not been added for legacy cases

File: client/app/queue/TaskSnapshot.jsx [hunk 3]
```diff
@@ -260,7 +261,13 @@ export class TaskSnapshot extends React.PureComponent<Props> {
...
                 </React.Fragment>
               }
-              {this.daysSinceTaskAssignmentListItem(task)}
+              { taskIsOnHold(task) ?
+                <React.Fragment>
+                  <dt>{COPY.CASE_LIST_TABLE_TASK_DAYS_ON_HOLD_COLUMN_TITLE}</dt>
+                  <dd><OnHoldLabel task={task} /></dd>
+                </React.Fragment> :
+                this.daysSinceTaskAssignmentListItem(task)
+              }
```
Comment (lowellrex) [inline client/app/queue/TaskSnapshot.jsx]: Let's move each item in the list out to its own function (or functional component) for tighter encapsulation and better scanability of the render function. I can imagine this part of the render function looking something like this:

```jsx
<td {...taskTimeContainerStyling}>
  <CaseDetailsDescriptionList>
    <AssignedOnListItem task={task} />
    <DueOnListItem task={task} />
    <OnHoldListItem task={task} />
  </CaseDetailsDescriptionList>
</td>
```

And one of those individual components could look something like:

```jsx
const OnHoldListItem = ({ task }) => {
  if (!taskIsOnHold(task)) {
    return daysSinceTaskAssignmentListItem(task);
  }

  return <React.Fragment>
    <dt>{COPY.CASE_LIST_TABLE_TASK_DAYS_ON_HOLD_COLUMN_TITLE}</dt>
    <dd><OnHoldLabel task={task} /></dd>
  </React.Fragment>;
};
```

**Comment by lowellrex** [comment]
Building some truth tables to help me understand what the changes will result from this PR.

### AMA appeals

. | Before | After | Parity?
--- | --- | --- | ---
Assigned to | [Assignee CSS ID](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56L140) | [Assignee CSS ID](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56R154) | ✅ 
Assigned by | [First initial, last name](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56L134) | [First initial, last name](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56R161) | ✅
Prepared by | [Latest attorney to draft a decision](https://github.com/department-of-veterans-affairs/caseflow/blob/c845fcda2a6029f10d8b19858727f090b9fd36b0/app/models/task.rb#L129) for the case's [first initial, last name](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56L136) | Latest attorney to draft a decision for the case's [first initial, last name](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56R168) | ✅
Type | [Task label](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56L151) | [Task label](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56R174) | ✅
Instructions | [Collapsable instructions](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56L159) | [Collapsable instructions](https://github.com/department-of-veterans-affairs/caseflow/pull/8480/files#diff-c6245bd098c838610d8b6f4dbce23d56R138) | ✅

Changes for AMA appeals look good!

File: client/app/queue/TaskSnapshot.jsx [hunk 4]
```diff
@@ -130,108 +129,77 @@ export class TaskSnapshot extends React.PureComponent<Props> {
...
+    return assignor ? <div><dt>{COPY.TASK_SNAPSHOT_TASK_FROM_LABEL}</dt>
+      <dd>{assignor}</dd></div> : null;
+  }
+
+  preparedByListItem = (task) => {
+    const preparedByAbbrev = task.decisionPreparedBy ? this.getAbbrevName(task.decisionPreparedBy) : null;
+
+    return preparedByAbbrev ? <div><dt>{COPY.TASK_SNAPSHOT_DECISION_PREPARER_LABEL}</dt>
+      <dd>{preparedByAbbrev}</dd></div> : null;
+  }
```
Comment (lowellrex) [inline client/app/queue/TaskSnapshot.jsx]: Let's get rid of the `preparedByListItem` function. This label shows the name of the attorney who drafted the decision. That information is now visible in the universal case title so we don't need to show it down here in the task as well (and this information relates more to the case than it does to the task).

**Pull request merged**